core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/ProtocolFact.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.protocols.json;
+
+import java.util.Collections;
+import java.util.Map;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.protocols.core.OperationInfo;
+import software.amazon.awssdk.utils.MapUtils;
+
+/**
+ * Records facts about each protocol.
+ */
+@SdkInternalApi
+public enum ProtocolFact {
```

Comment:
Should we use interface instead?

Reply:
We could but I like the enum just fine. Let me know if you feel strong about it and I can change it.

Reply:
Is there any benefit of using enum, it seems a bit unconventional to me to use enum for behavior